pkg/operator/ceph/cluster/controller.go

```diff
@@ -368,10 +368,23 @@ func (c *ClusterController) requestClusterDelete(cluster *cephv1.CephCluster) (r
 
 	logger.Infof("delete event for cluster %q in namespace %q", cluster.Name, cluster.Namespace)
 
-	if cluster, ok := c.clusterMap[cluster.Namespace]; ok && !cluster.closedStopCh {
-		// close the goroutines watching the health of the cluster (mons, osds, ceph status, etc)
-		close(cluster.stopCh)
-		cluster.closedStopCh = true
+	if cluster, ok := c.clusterMap[cluster.Namespace]; ok {
+		// if not already stopped, stop clientcontroller and bucketController
+		if !cluster.closedStopCh {
+			close(cluster.stopCh)
+			cluster.closedStopCh = true
+		}
+
+		// close the goroutines watching the health of the cluster (mons, osds, ceph status)
+		var isDisabled bool
+		for _, daemon := range monitorDaemonList {
+			isDisabled = isMonitoringDisabled(daemon, cluster.Spec)
```

Comment:
Will the `monitoringChannels` ever contain the channel if it is disabled? Seems like it should not be in the channels  if it's disabled, so the check for isDisabled would not be necessary.

Reply:
It should not have it if disabled but it might have it declared and then closed during the lifetime of the operator (let's enabled then disabled or the other way around). I'm just trying to avoid closing a nil/closed channel at this point. I believe this does not hurt.

Reply:
If there is an open channel, seems like we always need to disable it. If there is still a channel left open, that will be a bug. 

To avoid closing the channel again, we also can remove the closed channel from the map.

Separately, there are multiple goroutines accessing the map, so we need a lock around it.